packages/integrations/src/awslambda.ts

```diff
@@ -0,0 +1,255 @@
+       * process.env.AWS_LAMBDA_FUNCTION_NAME - this parameter provides the AWS Lambda Function name
+       */
+      return `https://${process.env.AWS_REGION}.console.aws.amazon.com/cloudwatch/home?region=${process.env.AWS_REGION}#logsV2:log-groups/log-group/$252Faws$252Flambda$252F${process.env.AWS_LAMBDA_FUNCTION_NAME}`;
+    };
+
+    /**
+     * This function sets Cloud Watch Logs data which are displayed in Sentry Dashboard
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+    const setCloudwatchLogsData = (scope: Scope): void => {
+      scope.setContext('cloudwatch logs', {
+        log_group: this._awsContext.logGroupName,
+        log_stream: this._awsContext.logStreamName,
+        url: cloudwatchUrl(),
+      });
+    };
+
+    /**
+     * This function sets tags which are displayed in Sentry Dashboard
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+
+    const setTags = (scope: Scope): void => {
+      scope.setTag('runtime', `node${global.process.version}`);
+      scope.setTag('transaction', this._awsContext.functionName);
+      scope.setTag('runtime.name', 'node');
+      scope.setTag('server_name', process.env._AWS_XRAY_DAEMON_ADDRESS || '');
+      scope.setTag('url', `awslambda:///${this._awsContext.functionName}`);
+    };
+
+    const flushTime = this._flushTime;
+    // timeout warning buffer for timeout error
+    const timeoutWarningBuffer: number = 1500;
+
+    const configuredTimeInSec = Math.floor(configuredTimeInMilliseconds / 1000);
+    const configuredTimeInMilli = configuredTimeInSec * 1000;
+
+    /**
+     * This function is invoked when there is timeout error
+     * Here, we make sure the error has been captured by Sentry Dashboard
+     * and then re-raise the exception
+     * @param configuredTime  - configured time in seconds
+     * @hidden
+     */
+    const timeOutError = (configuredTime: number): void => {
+      setTimeout(() => {
+        const error = new Error(
+          `WARNING : Function is expected to get timed out. Configured timeout duration = ${configuredTimeInSec +
+            1} seconds.`,
+        );
+
+        // setting parameters in scope which will be displayed as additional data in Sentry dashboard
+        hub.withScope((scope: Scope) => {
+          setTags(scope);
+          // runtime
+          setAdditionalRuntimeData(scope);
+          // setting the lambda parameters
+          setAdditionalLambdaParameters(scope);
+          // setting the cloudwatch logs parameter
+          setCloudwatchLogsData(scope);
+          // setting the sys.argv parameter
+          scope.setExtra('sys.argv', process.argv);
+          /** capturing the exception and re-directing it to the Sentry Dashboard */
+          hub.captureException(error);
+        });
+
+        /** capturing the exception and re-directing it to the Sentry Dashboard */
+        void hub.getClient()?.flush(flushTime);
```

Comment:
This `void` is unnecessary. Also if you cannot `await` here, then this `flush` has no point (unless AWS itself is waiting for the promises.

Reply:
Removing flush gives trouble for the timeout sync scenario, So we kept it.